lib/wda/webdriveragent.js

```diff
@@ -155,10 +155,27 @@ class WebDriverAgent {
     }
   }
 
-  async uninstall () {
-    log.debug(`Removing WDA application from device`);
+  /**
+   * Uninstall WDA from the test device.
+   * If WDA is built by Xcode 10-, the bundle id is WDA_BUNDLE_ID.
+   * If WDA is build by Xcode 11, the bundle id is 'com.facebook.WebDriverAgentRunner.xctrunner' for simulator.
+   * 'com.customised.WDA.xctrunner' for real device if a user sets 'updatedWDABundleId' as 'com.customised.WDA'.
+   *
+   * @param {string} updatedWDABundleId The bundle id which will be uninstalled
+   */
+  async uninstall (updatedWDABundleId = WDA_RUNNER_BUNDLE_ID) {
     try {
-      await this.device.removeApp(WDA_BUNDLE_ID);
+      if (this.xcodeVersion.major < 11) {
```

Comment:
it might be that xcode version is unknown for real devices

Reply:
ah...
Let me check again...

Reply:
`this.xcodeVersion` had Xcode version on a real device.
The value might be undefined if `this.opts.webDriverAgentUrl` was also provided in createSession

Reply:
> The value might be undefined if this.opts.webDriverAgentUrl was also provided in createSession

exactly